### PR TITLE
Fixup/settings page errors simple

### PR DESCRIPTION
**Pull request by crabsinger**
Fix bugs introduced in #1268 

- non of this should load on mainnet if flags arent enabled
- fix race condition bug vault selection

### CODE REVIEW

File: src/hooks/gql/useVaults.ts [hunk 1]
```diff
@@ -60,9 +65,9 @@ export function useVaults({
       return vaults;
     },
     {
-      enabled: !!orgId && !!chainId,
+      enabled: !!orgId && !!chainId && isFeatureEnabled('vaults'),
```
Comment (crabsinger) [inline src/hooks/gql/useVaults.ts:68]: is this necessary?

Comment (crabsinger) [reply src/hooks/gql/useVaults.ts:68]: tihink so.

**Review by wingfeatherwave — DISMISSED** [review]
nice work!

**Comment by wingfeatherwave** [comment]
🤝

**Review by wingfeatherwave — APPROVED** [review]
fo real this time

File: src/hooks/gql/useVaults.ts [hunk 2]
```diff
@@ -16,6 +18,9 @@ export function useVaults({
   return useQuery(
     ['vaults-for-org-', orgId],
     async () => {
+      if (!isFeatureEnabled('vaults')) {
```
Comment (levity) [inline src/hooks/gql/useVaults.ts:21]: this isn't redundant with the check in `enabled` below?

**Pull request merged**